include/ImmediateContext.inl

```diff
@@ -1078,6 +1084,12 @@ inline void ImmediateContext::PreDispatch() noexcept(false)
     PIXScopedEvent(GetGraphicsCommandList(), 0ull, L"PreDispatch");
     PreRender(COMMAND_LIST_TYPE::GRAPHICS);
 
+    if (m_bUseRingBufferDescriptorHeaps)
+    {
+        // Always dirty the state when using ring buffer heaps because we can't safely reuse tables in that case.

```

Comment:
In the original this was above PreRender

Reply:
Doesn't matter - PreRender only updates predication state, which isn't affected by these dirty flags. This was more aesthetic in my mind.